### PR TITLE
Test points functions

### PR DESCRIPTION
**Pull request by maxm17**

### CODE REVIEW

File: tests/unit_tests/grid/test_points_functions.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+    # Act
+    split_horizons_points = pf.split_horizons_points(basic_regular_grid)
+
+    # Assert
+    np.testing.assert_array_almost_equal(split_horizons_points, expected_split_horizons_points)
+
+
+def test_split_horizons_points_s_bend_faulted_grid(s_bend_faulted_grid):
+    # Act
+    split_horizons_points = pf.split_horizons_points(s_bend_faulted_grid)
```
Comment (andy-beer) [inline tests/unit_tests/grid/test_points_functions.py:406]: Please add a couple more tests using non zero values for (a) ref_k0 and (b) kp.  For the s_bend model, ref_k0 is in the range 0 to 4 I think.  kp may be 0 or 1.

File: tests/unit_tests/grid/test_points_functions.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    np.testing.assert_array_almost_equal(centre_points[1, 1, 4], np.array([1450., 2150., 3033.75]))
+    np.testing.assert_array_almost_equal(centre_points[2, 3, 5], np.array([1550., 2350., 3042.5]))
+
+
+# Need to check third test case
+@pytest.mark.parametrize("interpolation_fraction, expected_interpolated_point",
+                         [(np.array([0.0, 0.0, 0.0]), np.array([0.0, 0.0, 0.0])),
+                          (np.array([0.75, 0.75, 0.75]), np.array([75.0, 37.5, 15.0])),
+                          (np.array([0.75, 0.2, 0.1]), np.array([10.0, 10.0, 15.0]))])
+def test_interpolated_point(basic_regular_grid, interpolation_fraction, expected_interpolated_point):
```
Comment (andy-beer) [inline tests/unit_tests/grid/test_points_functions.py:338]: Please add a test like this using the s bend faulted grid, using a cell in the middle of the grid (not 0,0,0).

File: tests/unit_tests/grid/test_points_functions.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+                                                   [[1000., 2100., 3050.], [1000., 2200., 3050.]]],
+                                                  [[[1000., 2200., 3040.], [1000., 2300., 3040.]],
+                                                   [[1000., 2200., 3050.], [1000., 2300., 3050.]]],
+                                                  [[[1000., 2300., 3040.], [1000., 2400., 3040.]],
+                                                   [[1000., 2300., 3050.], [1000., 2400., 3050.]]],
+                                                  [[[1000., 2400., 3040.], [1000., 2500., 3040.]],
+                                                   [[1000., 2400., 3050.], [1000., 2500., 3050.]]]]])
+
+    # Act
+    x_section_corner_points = pf.x_section_corner_points(faulted_grid, axis = 'I')
```
Comment (andy-beer) [inline tests/unit_tests/grid/test_points_functions.py:150]: Please add a similar test for axis = 'J'

File: tests/unit_tests/grid/test_points_functions.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+                                                   [[0.0, 0.0, 20.0], [0.0, 50.0, 20.0]]],
+                                                  [[[0.0, 50.0, 0.0], [0.0, 100.0, 0.0]],
+                                                   [[0.0, 50.0, 20.0], [0.0, 100.0, 20.0]]]],
+                                                 [[[[0.0, 0.0, 20.0], [0.0, 50.0, 20.0]],
+                                                   [[0.0, 0.0, 40.0], [0.0, 50.0, 40.0]]],
+                                                  [[[0.0, 50.0, 20.0], [0.0, 100.0, 20.0]],
+                                                   [[0.0, 50.0, 40.0], [0.0, 100.0, 40.0]]]]])
+
+    # Act
+    x_section_corner_points = pf.x_section_corner_points(basic_regular_grid, axis = 'I')
```
Comment (andy-beer) [inline tests/unit_tests/grid/test_points_functions.py:98]: Please add a similar test for axis = 'J' (in that case the x values should vary, with y values constant).

**Review by andy-beer — COMMENTED** [review]
All good as far as I can see.  Please add a few more tests as indicated.

**Comment by maxm17** [comment]
@andy-beer Added the tests asked for. If you would like to check them then merge if they are fine

**Review by andy-beer — APPROVED** [review]
